egs/multi_en/s5/local/g2p/train_g2p.sh

```diff
@@ -1,83 +1,61 @@
 #!/bin/bash
 
-###########################################################################################
-# This script was copied from egs/librispeech/s5/local/g2p/train_g2p.sh
-# The source commit was e69198c3dc5633f98eb88e1cdf20b2521a598f21
-# Changes made:
-# - Removed CMUDict download/cleaning steps
-# - Changed to use data/local/dict_nosp/lexicon.txt instead
-# - Renumbered stages
-###########################################################################################
+# Copyright 2017  Intellisist, Inc. (Author: Navneeth K)
+#           2017  Xiaohui Zhang
+# Apache License 2.0
 
-# Copyright 2014 Vassil Panayotov
-# Apache 2.0
+# This script trains a g2p model using Phonetisaurus and SRILM.
 
-# Trains Sequitur G2P models on CMUdict
+stage=0
+silence_phones=
 
-# can be used to skip some of the initial steps
-stage=1
+echo "$0 $@"  # Print the command line for logging
 
-. utils/parse_options.sh || exit 1
-. ./path.sh || exit 1
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. utils/parse_options.sh || exit 1;
 
-if [ $# -ne "2" ]; then
-  echo "Usage: $0 <train-lexicon> <g2p-dir>"
-  exit 1
+
+if [ $# -ne 2 ]; then
+	echo "Usage: $0 <dictdir> <outdir>"
+	exit 1;
 fi
 
-train_lex=$1
-g2p_dir=$2
+lexicondir=$1
+outdir=$2
+
+[ ! -f $lexicondir/lexicon.txt ] && echo "Cannot find $lexicondir/lexicon.txt" && exit
 
-mkdir -p $g2p_dir
+mkdir -p $outdir
 
-model_1=$g2p_dir/model-1
-model_2=$g2p_dir/model-2
-model_3=$g2p_dir/model-3
-model_4=$g2p_dir/model-4
-model_5=$g2p_dir/model-5
 
-if [ -f $model_5 ]; then
-  echo "$model_5 already exists. Skipping G2P model training..."
-  exit 0;
+# For input lexicon, Remove pronunciations containing non-utf-8-encodable characters,
+# and optionally remove words that are mapped to a single silence phone from the lexicon.
+if [ $stage -le 0 ]; then
+  lexicon=$lexicondir/lexicon.txt
+  if [ ! -z "$silence_phones" ]; then
+    awk 'NR==FNR{a[$1] = 1; next} {s=$2;for(i=3;i<=NF;i++) s=s" "$i; if(!(s in a)) print $1" "s}' \
+      $silence_phones $lexicon | \
+      awk '{printf("%s\t",$1); for (i=2;i<NF;i++){printf("%s ",$i);} printf("%s\n",$NF);}' | \
+      iconv -c -t utf-8 -  | awk 'NF > 0'> $outdir/lexicon_tab_separated.txt
+  else
+    awk '{printf("%s\t",$1); for (i=2;i<NF;i++){printf("%s ",$i);} printf("%s\n",$NF);}' $lexicon | \
+      iconv -c -t utf-8 -  | awk 'NF > 0'> $outdir/lexicon_tab_separated.txt
```

Comment:
IIRC (and I might be mistaken), `-c` causes removal of the individual (faulty) characters, not the full words. Is this really what you want?  Do you have an example of words/characters for which it happens?

Reply:
Thanks Yenda for pointing out. That's indeed a problem. I want to ignore lines with non-utf8-encodable characters like andrikienė, in G2P training data. What's the more straightforward way to do it in your opinions? Thanks!

Reply:
@jtrmal, do you have suggestion here? thanks

Reply:
sorry, I forgot about this issue -- let me look at it and get back to you.

Reply:
I think you could do something like
```
iconv --unicode-subst='{INVALID}' -t utf-8 - | grep -v -F ''{INVALID}'
```

might take some experimenting to do it correctly.